nightly_upload.sh

```diff
@@ -0,0 +1,39 @@
+#!/bin/bash
+# nightly_upload: checks remote file for changes and uploads new version if necessary
+# Usage:
+# set scp(ssh) server with user and path
+# set ssh port
+# create env-variable with private key: $ssh_server_key
+# call script: ./nightly_upload.sh -filename
+
+repo_name=$(basename `git rev-parse --show-toplevel`)
+
+ssh="drupal@www.nitrokey.com:~/www/nitrokey.com/files/ci/"$repo_name'/' #-P 2215 drupal@www.nitrokey.com:~/www/nitrokey.com/files/ci/pynitro>
+port=2215
+build_file="$1"
+new_sha=$build_file"-sha256sum.txt"
+
+old_sha="old"-$new_sha
+date_now=$(date "+%F-%H-%M-%S")
+echo $repo_name
+
+# create ssh-key file
+echo "-----BEGIN OPENSSH PRIVATE KEY-----" > ~/server_key
+echo "$ssh_server_key" >> ~/server_key
+echo "-----END OPENSSH PRIVATE KEY-----" >> ~/server_key
```

Comment:
Is there no better way to transport the SSH key, except than using variables?
Alternatively we could store it encrypted, and pass decryption keys or use external service to run the decryption.
Or even better, we could think about using the HSM for that.

Reply:
on the last call we decided to leave it like this for now and use NetHSM as soon as it is ready